### DISCUSSION TITLE
Collision ends if actor isn't actively moving into collider...

### DISCUSSION THREAD

**Discussion by brettkoz** [excaliburjs/Excalibur#3206] [Q&A]
I'm building a beekeeping game where you have to walk around your bee yard and interact with your hives. As you walk around the yard, you should have the ability to see a popup of hive information when you collide with a hive. The problem is, I have to have collisions set to passive for the info window to stay up, otherwise, with collisions active I have to be actively walking into the hive to keep the information window up... My question is, is there a way to allow a buffer zone for collisions, where the onCollisionEnd doesn't get called immediately upon releasing the direction key?

**Comment by eonarheim** [comment]
@brettkoz Thanks for the question! Sorry I missed this before your discord post. 

For future explorers from google we answered it here https://discord.com/channels/1195771303215513671/1283888966231396443/1283888966231396443

@mattjennings Suggested using a child actor as a sensor

> hm, i would add an invisible, passive actor as a child to the hive that acts as the "buffer zone", then listen for collision events from the player on that.
> https://excaliburjs.com/docs/collision-events/#collision-start-collisionstart docs for the collision events
> ```typescript
> // typing this off the top of my head in discord so some things might be wrong
> 
> 
> class Beehive extends ex.Actor {
>    buffer: ex.Actor
> 
>    constructor(args) {
>       super(args)
> 
>       this.buffer = new ex.Actor({ 
>          radius: 10, 
>          collisionType: ex.CollisionType.Passive,
>          // .. whatever else as necessary
>       })
> 
> 
>       this.addChild(this.buffer)
> 
>       this.buffer.on('collisionstart', ev => {
>          // if collision is with player, show label
>       })
> 
>       this.buffer.on('collisionend', ev => {
>          // if player stopped colliding, hide label
>       })
>       
>    }
> }
> 
> ```
>
